### ISSUE TITLE
Type definition emitted for destructuring parameter is considered invalid by compiler

### ISSUE THREAD

**Issue by RIAstar** [Microsoft/TypeScript#9644]
This problem looks very much like #3912, but it's not a duplicate.
The difference is that here we're not emitting an interface, but an anonymous function.

I think this is a bug because it's valid ES6 code that results in breaking compilation for unexpected reasons.

**TypeScript Version:**  1.8.10

**Code**
```ts
export const test = ({id}) => ({id});
```
This is valid ES6 and emits:
```ts
export declare const test: ({id}: {id: any;}) => {id: any;};
```
I would expect this to be valid, but it's not. The compiler doesn't accept it.

In contrast a named function emits something very similar, that the compiler accepts gladly:
```ts
export function test({id}) {return {id};}
```
emits
```ts
export declare function test({id}: {id: any;}): {id: any;};
```
So exactly the same but for the keyword (const vs. function) and the fat arrow.

----
Unfortunately, it's not much better with explicit types:
```ts
export interface Foo {
    id: number;
}

export const test = ({id}:Foo):Foo => ({id});
```

yields
```ts
export interface Foo {
    id: number;
}
export declare const test: ({id}: Foo) => Foo;
```
which the compiler also doesn't like.

---
**Workaround** (besides using a named function)
Type the function with an interface:
```ts
export interface Test {
    ({id});
}
export const test:Test = ({id}) => ({id});
```
Yields:
```ts
export interface Test {
    ({id}: {id: any;}): any;
}
export declare const test: Test;
```
which is accepted.

**Comment by mhegazy** [comment]
This is fixed in TS 2.0-beta. please give it a try.

**Comment by mhegazy** [comment]
i do not think there is a reasonable work around for TS 1.8 for this other than specifying an explicit type annotation on your variable declarations.
